androidApp/src/main/java/com/example/tasky/android/login/viewmodel/RegisterViewModel.kt

```diff
@@ -4,13 +4,19 @@ import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.example.tasky.android.login.screen.RegisterScreenEvent
 import com.example.tasky.android.login.screen.RegisterScreenState
+import com.example.tasky.model.login.RegisterBody
+import com.example.tasky.model.onError
+import com.example.tasky.model.onSuccess
+import com.example.tasky.repository.LoginRepository
 import com.example.tasky.util.Validator
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 
 class RegisterViewModel : ViewModel() {
+    private val loginRepository = LoginRepository()
+
     private val _screenStateFlow = MutableStateFlow(RegisterScreenState())
     val screenStateFlow = _screenStateFlow.asStateFlow()
 
```

Comment:
If you use the delegate here, you can avoid needing two properties:
```
var state by mutableStateOf(RegisterScreenState())
    private set
```

Reply:
can you explain the pros and cons between MutableStateFlow and MutableState? As I know both of them can be used as state in composable, but in general MutableStateFlow is recommended since it is not binded with composable and provides more flexibility.